src/rhi-shared.cpp

```diff
@@ -61,6 +67,50 @@ Result Buffer::getSharedHandle(NativeHandle* outHandle)
     return SLANG_E_NOT_AVAILABLE;
 }
 
+// ----------------------------------------------------------------------------
+// Texture helpers
+// ----------------------------------------------------------------------------
+
+Result calcSubresourceRegionLayout(
+    const TextureDesc& desc,
+    uint32_t mipLevel,
+    uint32_t layerIndex,
+    Offset3D offset,
+    Extents extents,
+    Size rowAlignment,
+    SubresourceLayout* outLayout
+)
+{
+    Extents textureSize = desc.size;
+    const FormatInfo& formatInfo = getFormatInfo(desc.format);
+
+    if (extents.width == kRemainingTextureSize)
```

Comment:
would it make sense to first compute the full extent of the texture, then check that offset is in range, and then shrink the extent? this will overflow if any(offset > extent)